test/helpers/kubectl.go

```diff
@@ -1130,3 +1130,102 @@ func (kub *Kubectl) GetCiliumPodOnNode(namespace string, node string) (string, e
 
 	return res.Output().String(), nil
 }
+
+// CiliumPreFlightCheck specify that it checks that various subsystems within
+// Cilium are in a good state. If one of the multiple preflight fails it'll
+// return an error.
+func (kub *Kubectl) CiliumPreFlightCheck() error {
+	// Doing this withTimeout because the Status can be ready, but the other
+	// nodes cannot be show up yet, and the cilium-health can fail as a false positive.
+	var err error
+	body := func() bool {
+		err = kub.CiliumControllersPreFlightCheck()
+		if err != nil {
+			return false
+		}
+		err = kub.CiliumHealthPreFlightCheck()
+		if err != nil {
+			return false
+		}
+		return true
+	}
+	timeoutErr := WithTimeout(body, "PreflightCheck failed", &TimeoutConfig{Timeout: HelperTimeout})
+	if timeoutErr != nil {
+		return fmt.Errorf("CiliumPreFlightCheck error: %s: %s", timeoutErr, err)
+	}
+	return nil
+}
+
+// CiliumControllersPreFlightCheck validates that all controllers are not
+// failing. If any of the controllers fails will return an error.
+func (kub *Kubectl) CiliumControllersPreFlightCheck() error {
+	var controllersFilter = `{range .controllers[*]}{.name}{"="}{.status.consecutive-failure-count}{"\n"}{end}`
+	ciliumPods, err := kub.GetCiliumPods(KubeSystemNamespace)
+	if err != nil {
+		return err
+	}
+	for _, pod := range ciliumPods {
+		status := kub.CiliumExec(pod, fmt.Sprintf(
+			"cilium status --all-controllers -o jsonpath='%s'", controllersFilter))
+		if !status.WasSuccessful() {
+			return fmt.Errorf("cilium-agent %q: Cannot run cilium status: %s",
+				pod, status.OutputPrettyPrint())
+		}
+		for controller, status := range status.KVOutput() {
+			if status != "0" {
+				failmsg := kub.CiliumExec(pod, "cilium status --all-controllers")
+				return fmt.Errorf(
```

Comment:
Do we want to show show all failing controllers here? If yes, we would need to aggregate these errors into a higher level error or log them in tests and fail at the end of tests.

Reply:
No, we don't at all. All that logs are saved in the `test-output.log` if further debugging is needed. 

If we dump that, the output of the Junit will be massive and not useful for users.

Reply:
I think part of the point of making these changes is to have a more relevant summary output shown directly in the jenkins UI (so in the junit output). In particular, it's a little burdensome to read through the full log for something in the middle. We have a lot of checks now, so you would expect a lot of logs.
In most cases, the preflight will work so it doesn't matter. If a single controller fails, the code, as-is, would print it. If a group of controllers fails, however, nebril's suggestion is relevant. If I look at the output and see 1 CNP sync controller fail, I know it was specific to that controller, if I see that all of them have failed I know it's more systemic. Similarly, seeing all controllers related to one endpoint fails says something different than if only one did. I think I also prefer printing all of them. In all cases, the preflight checks would block more code running so they won't flood the output (ideally we would only print anything on error).

Another issue with printing only one error is that the order of the `KVOutput` output might be different between runs. This might be confusing. Whether we agree to print all the failing controllers or not, we should try to sort the `KVOutput` output if we can, so we always print the same error.

Reply:
I chatted with eloycoto offline and `failmsg.OutputPrettyPrint()` will actually pretty print all the controllers.